geotrek/sensitivity/views.py

```diff
@@ -116,75 +139,133 @@ class SensitiveAreaAPIViewSet(APIViewSet):
-        ))
-        # Ensure smaller areas are at the end of the list, ie above bigger areas on the map
-        # to ensure we can select every area in case of overlapping
-        qs = qs.annotate(area=Area('geom2d_transformed')).order_by('-area')
-
-        if 'practices' in self.request.GET:
-            qs = qs.filter(species__practices__name__in=self.request.GET['practices'].split(','))
+        qs = (
+            (
+                SensitiveArea.objects.existing()
+                .filter(published=True)
+                .select_related("species", "structure")
+                .prefetch_related(
+                    "species__practices",
+                    Prefetch(
+                        "attachments",
+                        queryset=Attachment.objects.select_related(
+                            "license", "filetype", "filetype__structure"
+                        ),
+                    ),
+                )
+            )
+            .annotate(geom_type=GeometryType(F("geom")))
+            .annotate(
+                geom2d_transformed=Case(
+                    When(
+                        geom_type="POINT",
+                        then=Transform(
+                            Buffer(F("geom"), F("species__radius"), 4),
+                            settings.API_SRID,
+                        ),
+                    ),
+                    When(
+                        geom_type__in=("POLYGON", "MULTIPOLYGON"),
+                        then=Transform(F("geom"), settings.API_SRID),
+                    ),
+                )
+            )
+            .annotate(area=Area("geom2d_transformed"))
+            .order_by("-area")
+        )
+
+        if "practices" in self.request.GET:
+            qs = qs.filter(
+                species__practices__name__in=self.request.GET["practices"].split(",")
+            )
 
         return qs
 
 
-if 'geotrek.trekking' in settings.INSTALLED_APPS:
+if "geotrek.trekking" in settings.INSTALLED_APPS:
+
     class TrekSensitiveAreaViewSet(viewsets.ModelViewSet):
         model = SensitiveArea
         serializer_class = SensitiveAreaAPIGeojsonSerializer
         permission_classes = [rest_permissions.DjangoModelPermissionsOrAnonReadOnly]
 
         def get_queryset(self):
-            pk = self.kwargs['pk']
+            pk = self.kwargs["pk"]
             trek = get_object_or_404(Trek.objects.existing(), pk=pk)
             if not trek.is_public():
                 raise Http404
             qs = trek.published_sensitive_areas
-            qs = qs.prefetch_related('species')
-            qs = qs.annotate(geom_type=GeometryType(F('geom')))
-            qs = qs.annotate(geom2d_transformed=Case(
-                When(geom_type='POINT', then=Transform(Buffer(F('geom'), F('species__radius'), 4), settings.API_SRID)),
-                When(geom_type='POLYGON', then=Transform(F('geom'), settings.API_SRID))
-            ))
+            qs = qs.prefetch_related("species")
```

Comment:
Please avoid reformatting this whole file

Reply:
Which formatters (python and jinja/html/js) to use and where can i find formatter config in this project ? These changes occured by using autopep8.